app/controllers/split_checkout_controller.rb

```diff
@@ -16,10 +16,14 @@ class SplitCheckoutController < ::BaseController
   helper OrderHelper
 
   def edit
-    redirect_to_step unless params[:step]
+    return redirect_to_step unless params[:step]
+
+    return redirect_to_guest if !@order.distributor.allow_guest_orders? && params[:step] != "guest"
```

Comment:
The logic here seems not straightforward. Should `redirect_to_guest` rather be named `redirect_to_guest_or_login`?

Reply:
Maybe.
The point here is that the step itself is named `guest` (and then the view is also named `guest`). And this method redirect to that URL/view, that's why.
You're right, it's more `not logged`, `disconnected`, `anonymous` but I guess it's not that beautiful in the url ... If you have any proposals?! 🙏

Reply:
If I understand well, the "guest view" the user is redirected to won't display guest but only the login option in this case, as guest is disabled is this distributor. So I think it would be clearer if the step and view were named something like "authentication" (with two choices guest/anonymous and logged_in_user). But I suppose it's quite out of the scope of this PR... :D

Reply:
`redirect_to_guest` is also used to redirect to `/checkout/guest` when distributor allow guest order, ie. a view when you can see login and checkout as user:
<img width="1026" alt="Capture d’écran 2022-01-05 à 16 34 27" src="https://user-images.githubusercontent.com/296452/148528951-acf8cf84-fcdd-47ca-8eb4-16542f590305.png">


I can rename it to `authentication` if it's much clear for everyone ;-)